dss/api/files.py

```diff
@@ -121,12 +120,22 @@ def get_helper(uuid: str, replica: Replica, version: str = None, token: str = No
     return response
 
 
-def _verify_checkout(
-        replica: Replica, token: typing.Optional[str], file_metadata: dict, blob_path: str,
-) -> typing.Tuple[str, bool]:
+def _verify_checkout(replica: Replica, file_metadata: dict, blob_path: str,) -> typing.Tuple[str, bool]:
     cloud_handle = Config.get_blobstore_handle(replica)
     hca_handle = Config.get_hcablobstore_handle(replica)
 
+    # Lookup existing checkout job. token is an exception if checkout failed
+    async_key = f"{replica.name}/{blob_path}"
+    token = AsyncStateItem.get(async_key)
```

Comment:
There are multiple places within this PR that multiple small changes are scattered through and across various functions/methods. Doing so is expedient for testing implementation ideas, yet it can make potentially make the code brittle as it is evolved by other authors. IMO, it would make the code more maintainable to have a helper class, perhaps `CheckoutTracker`, that provides a well-defined interface and encapsulates the details of key formation, `AsynchStateItem` use, etc.